runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpStateSingle.java

```diff
@@ -182,6 +218,12 @@ boolean update(int currentXp, int goalStartXp, int goalEndXp)
 		int originalXp = xpGained + startXp;
 		int actionExp = currentXp - originalXp;
 
+		//If we are currently paused we dont want to update
+		if (isPaused())
+		{
+			xpPaused += actionExp;
```

Comment:
This is getting way too complicated.

Reply:
You would still need not subtract it from the xp gained as the way it is currently tracked is by subtracting the starting experience from the current experience. When you were to unpause it would simply add the paused experience in one chunk to the gained experience. I didn't want to mess with the current code too much and I don't think this solution is really all that complicated.

Reply:
It muddies what variables are what.

I still want xpGained to update on every XP event, even if it isn't sent to the view.

Reply:
It is purely an internal representation. I could change the variable name to represent more actually what it stands for. None of the classes requesting this data will do it by directly accessing the fields. Would this be a fair solution to you or would you rather change the way xp gained is calculated on xp events (this would however mean more is modified from the current working version of course).

Reply:
I find this current implementation to obscure the behavior given the plugin's internal state. 

I think making it clear of "here's how it behaves when it is paused" will resolve my concern here.

Reply:
I count the experience gained when the skill tracker for the specific skill is currently paused. This paused exp is subtracted from the total gained exp only when a snapshot is requested. This makes it so that none of the functionality from before the pause function changes because it only adds a field and changes the value(s) returned in the snapshot(s).

Reply:
I don’t doubt the correctness in operation. I understand it now.

I also understand that what I propose will take more lines.

Just that this current implementation is not obviously comprehensible, is what I am critiquing.

Reply:
I'm not sure that I agree; maybe xpGained should be named more clearly but besides that I'm not sure why the implementation wouldn't be obviously comprehensible.